### PR TITLE
refactor: quotes types (zaitsev-oleksii)

### PR DESCRIPTION
**Pull request by zaitsev-oleksii**
### Checks

- [ ] Adding quotes?
  - [ ] Make sure to include translations for the quotes in the description (or another comment) so we can verify their content.
- [ ] Adding a language or a theme?
  - [ ] If is a language, did you edit `_list.json`, `_groups.json` and add `languages.json`?
  - [ ] If is a theme, did you add the theme.css?
    - Also please add a screenshot of the theme, it would be extra awesome if you do so!
- [x] Check if any open issues are related to this PR; if so, be sure to tag them below.
- [x] Make sure the PR title follows the Conventional Commits standard. (https://www.conventionalcommits.org for more info)
- [x] Make sure to include your GitHub username inside parentheses at the end of the PR title

related to https://github.com/monkeytypegame/monkeytype/issues/4786

### CODE REVIEW

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/7703190965

File: frontend/src/ts/ape/types/quotes.d.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  type GetQuotes = Quote[];
+  type GetIsSubmissionEnabled = { isEnabled: boolean };
+  type GetRating = QuoteRating | null;
+
+  type PostQuotes = void;
+  type PostApprove = ApproveReturn;
+  type PostReject = void;
+  type PostRating = QuoteRating | null;
+  type PostReport = void;
```
Comment (Miodec) [inline frontend/src/ts/ape/types/quotes.d.ts:40]: Replace `void` with `null`

**Review by Miodec — CHANGES_REQUESTED** [review]

File: frontend/src/ts/ape/types/ape.d.ts [hunk 2]
```diff
@@ -13,18 +13,17 @@ declare namespace Ape {
...
+  type HttpMethodTypes = "get" | "post" | "put" | "patch" | "delete";
 
-  type HttpMethodTypes = keyof HttpClient;
+  type HttpClient = {
+    [Method in HttpMethodTypes]: <Data = unknown>(
+      endpoint: string,
+      config?: RequestOptions
+    ) => EndpointResponse<Data>;
+  };
 }
```
Comment (Miodec) [inline frontend/src/ts/ape/types/ape.d.ts:29]: Also, why are these changes needed?

Comment (zaitsev-oleksii) [reply frontend/src/ts/ape/types/ape.d.ts:29]: Without making HttpClient methods generic getting type errors here
Do you have any ideas on how I can fix those in a better way?

image

Comment (Miodec) [reply frontend/src/ts/ape/types/ape.d.ts:29]: image

Comment (zaitsev-oleksii) [reply frontend/src/ts/ape/types/ape.d.ts:29]: okay, thank you, will make changes

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/7717215110

**Review by Miodec — APPROVED** [review]
Thanks

**Pull request merged**